### PR TITLE
Allow to export select pixels

### PR DESCRIPTION
**Pull request by willdurand**

### CODE REVIEW

File: apps/explorer/views.py [hunk 1]
```diff
@@ -1,15 +1,32 @@
...
+def str_to_set(input):
+    """Returns a set of strings by splitting the given `input` string on space,
+    comma or new line. It eliminates duplicates and strips each string.
+    """
+    return set(
+        filter(
+            None,
+            [part.strip() for part in re.split('\s*,\s*|\s+|\n', input)]
+        )
+    )
```
Comment (jmaupetit) [inline apps/explorer/views.py]: I think `filter(None, iterable)` is harder to read/understand than the following:

```python
set([part.strip() for part in re.split('\s*,\s*|\s+|\n', input) if part])
```

WDYT?

I have to confess that I wasn't aware of this `None` function trick.

File: apps/explorer/views.py [hunk 2]
```diff
@@ -150,5 +167,43 @@ def get_context_data(self, **kwargs):
         context.update({
             'pixels': pixels,
             'pixels_limit': self.pixels_limit,
+            'export_form': PixelSetExportPixelsForm(),
         })
         return context
+
+
+class PixelSetExportPixelsView(LoginRequiredMixin, SingleObjectMixin,
+                               FormView):
```
Comment (jmaupetit) [inline apps/explorer/views.py]: I would prefer the following indentation:

```python
class PixelSetExportPixelsView(LoginRequiredMixin, 
                               SingleObjectMixin,
                               FormView):
```

File: apps/explorer/templates/explorer/pixelset_detail.html [hunk 3]
```diff
@@ -149,11 +153,10 @@ <h4>{% trans "Subset selection" %}</h4>
         </p>
         <button
           type="submit"
-          name="select"
           class="button"
-          disabled
         >
-          {% trans "Select" %}
+          <i class="fa fa-download" aria-hidden="true"></i>
+          {% trans "Download an archive (.zip) with the selected Pixels" %}
```
[the remark this answers is not in the document]
Comment (willdurand) [reply apps/explorer/templates/explorer/pixelset_detail.html]: ah oops

File: apps/explorer/tests/test_views.py [hunk 4]
```diff
@@ -12,7 +13,8 @@
 from apps.core.management.commands.make_development_fixtures import (
     make_development_fixtures
 )
-from apps.explorer.views import PixelSetDetailView, PixelSetExportView
+from apps.explorer.views import (PixelSetDetailView, PixelSetExportView,
+                                 PixelSetExportPixelsView, str_to_set)
```
[the remark this answers is not in the document]
Comment (willdurand) [reply apps/explorer/tests/test_views.py]: if you say so..

File: apps/explorer/views.py (hunk 1 above)
Comment (willdurand) [reply apps/explorer/views.py]: It does not do the same thing, I want filtering after `strip()`

Comment (willdurand) [reply apps/explorer/views.py]: I added a comment

File: apps/explorer/tests/utils/test_export.py [hunk 5]
```diff
@@ -207,21 +207,40 @@ def test_export_empty_pixelset(self):
...
+
+        pixels_csv = self._export_pixels(
+            pixel_set,
+            omics_units=[selected_pixel.omics_unit.reference.identifier]
+        )
+
+        assert len(pixels_csv['Omics Unit'].items()) == 1
+
+        assert (pixels_csv['Omics Unit'][0] ==
+                selected_pixel.omics_unit.reference.identifier)
```
[the remark this answers is not in the document]
Comment (willdurand) [reply apps/explorer/tests/utils/test_export.py]: I don.t think it is worth declaring variables jus for having single lines < 79 chars...

File: apps/explorer/forms.py [hunk 6]
```diff
@@ -59,3 +59,8 @@ class PixelSetExportForm(forms.Form):
             'required': _('You must select at least one Pixel Set.'),
         },
     )
+
+
+class PixelSetExportPixelsForm(forms.Form):
+
+    omics_units = forms.CharField(required=True)
```
[the remark this answers is not in the document]
Comment (willdurand) [reply apps/explorer/forms.py]: I don't think we can do anything, we get a string as input and we split it. How can you determine what is wrong based on that?

File: apps/explorer/tests/utils/test_export.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (jmaupetit) [reply apps/explorer/tests/utils/test_export.py]: I understand your concern and I won't insist on this.

File: apps/explorer/views.py (hunk 1 above)
Comment (jmaupetit) [reply apps/explorer/views.py]: You are right: I missed the point.

File: apps/explorer/forms.py (hunk 6 above)
[the remark this answers is not in the document]
Comment (jmaupetit) [reply apps/explorer/forms.py]: We can postpone this to a later improvement.

**Pull request merged**